app/core/hooks/useSearch.ts

```diff
@@ -0,0 +1,22 @@
+import {useEffect, useState} from "react"
+import {useDispatch} from "react-redux"
+import {search} from "src/js/flows/search/mod"
+import {AppDispatch} from "src/js/state/types"
+import {zng} from "zealot"
+
+type R = [zng.Record[], boolean]
+
+export default function useSearch(query: string, deps?: any[]): R {
+  const dispatch = useDispatch<AppDispatch>()
+  const [records, setRecords] = useState<zng.Record[]>([])
+  const [isFetching, setIsFetching] = useState<boolean>(false)
+
+  useEffect(() => {
+    const {response, abort} = dispatch(search({query}))
+    response.chan(0, (records) => setRecords(records))
+    response.status((status) => setIsFetching(status === "FETCHING"))
+    return abort
+  }, deps)
+
+  return [records, isFetching]
+}
```

Comment:
I think this will be the future of data fetching in the dashboard widgets. It's a simple hook that returns the records as they come in, and a boolean isFetching. 

```js
const [records, isFetching] = useSearch(* | count() by _path", [])
```